### ISSUE TITLE
Feature Request - Concat - Provide Chapter Marks to timestamp concat'd videos

### ISSUE THREAD

**Issue by Zefferis** [yt-dlp/yt-dlp#2437]
### Checklist

- [X] I'm reporting a feature request
- [X] I've verified that I'm running yt-dlp version **2022.01.21**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

## Request:
Add an option to extend --concat-playlist to create Chapter titles like "video 1, video 2, video 3, ect" to help distinguish the individual videos in the new single larger file. 

Something like ``--concat-playlist-make-chapters`` ?

The name of the chapter marks is less of a concern than just having an entry in chapters to help delineate when a playlist video starts and ends within the larger concat'd playlist.

Thanks!

**Comment by chrizilla** [comment]
@Zefferis : good idea. Question: Can a video be concatted gaplessly and error-free(ly) in this manner?
In my experience, the resulting video and/or audio track suffer from gaps during playback.

**Comment by Zefferis** [comment]
> @Zefferis : good idea. Question: Can a video be concatted gaplessly and error-free(ly) in this manner?
> In my experience, the resulting video and/or audio track suffer from gaps during playback.

Effectly were talking about taking the duration of the video and making a chapter file that has those timestamps, - then adding the duration of the next file ontop for the next spot in the chapter file, ect ect

This feature exists already for single videos with chapters.

This is talking about treating each video duration itself as a chapter within the complete concat

**Comment by chrizilla** [comment]
@Zefferis : I'm not at all critizing your idea, which I think is a good one. I was merely wondering whether the concatenation produces a gapless and errorfree audio/video file. When I play concatenated files like this in VLC or PotPlayer, there are always gaps in the audio and jumps in the video track at the concatenation points. (That's why I hesitate to concatenate the parts/chapters into a faulty single file). I just wanted to ask if you have the same or a better experience than myself (seeing that you apparently in the habit of concatenating videos).

**Comment by Zefferis** [comment]
> @Zefferis : I'm not at all critizing your idea, which I think is a good one. I was merely wondering whether the concatenation produces a gapless and errorfree audio/video file. When I play concatenated files like this in VLC or PotPlayer, there are always gaps in the audio and jumps/artefacts in the video track at the concatenation points. That's why I hesitate to concatenate the parts/chapters into a faulty single file. I just wanted to ask if you have the same or a better experience than myself (as you seem to be in the habit of concatenating videos).

No worries! just used it as a chance to explain more of what I was thinking.

I'm using MPV to play these files and they're working fine for me rn, but I'm using them on YouTube playlist that are series' from the same creator, so no issues with varied audio or video files so far with my experience

**Comment by pukkandan** [comment]
If you are using the same format for all videos, the transitions should be seamless. Note that using the same "format selector" does NOT guarantee the same "format" across videos. While I generally don't recommend it, if you are concating videos, I would suggest selecting formats by id to ensure consistency
